lib/fetch.go

```diff
@@ -64,46 +65,48 @@ func (of *OCIFetcher) debugMsg(format string, a ...interface{}) {
 }
 
 func blobsDir(outputDir string) string {
-	return path.Join(outputDir, "blobs")
+	return filepath.Join(outputDir, "blobs")
 }
 
 func blobFile(outputDir string, digest string) string {
 	formattedDigest := strings.Replace(digest, ":", "-", -1)
-	return path.Join(blobsDir(outputDir), formattedDigest)
+	return filepath.Join(blobsDir(outputDir), formattedDigest)
 }
 
 func refsDir(outputDir string) string {
-	return path.Join(outputDir, "refs")
+	return filepath.Join(outputDir, "refs")
 }
 
 // Fetch will download the image represented by u into outputDir.
 func (of *OCIFetcher) Fetch(u *URL, outputDir string) error {
-	// fetch the manifest and config
-	of.debugMsg("fetching OCI image host:%s, name:%s, tag:%s", u.Host, u.Name, u.Version)
-	manifest, err := of.fetchManifest(u)
+	// create the blobs and refs directories
+	err := os.MkdirAll(blobsDir(outputDir), 0755)
 	if err != nil {
 		return err
 	}
-	of.debugMsg("manifest successfully retrieved")
-	config, err := of.fetchConfig(u, manifest.Config.Digest, manifest.Config.Size)
+	err = os.MkdirAll(refsDir(outputDir), 0755)
 	if err != nil {
 		return err
 	}
-	of.debugMsg("config successfully retrieved")
 
-	// create the blobs and refs directories
-	err = os.MkdirAll(blobsDir(outputDir), 0755)
+	// fetch the manifest and config
+	of.debugMsg("fetching OCI image host:%s, name:%s, tag:%s", u.Host, u.Name, u.Version)
+	manifestData, err := of.fetchManifest(u, outputDir)
 	if err != nil {
 		return err
 	}
-	err = os.MkdirAll(refsDir(outputDir), 0755)
+	manifest := manifestData.manifest
+	of.debugMsg("manifest successfully retrieved")
+
+	configData, err := of.fetchConfig(u, manifest.Config.Digest, manifest.Config.Size, outputDir)
 	if err != nil {
 		return err
 	}
+	of.debugMsg("config successfully retrieved")
 
 	// download all of the layers into the blobs directory, displaying progress
 	// bars for the user
-	cpp := &progressutil.CopyProgressPrinter{}
+	cpp := progressutil.NewCopyProgressPrinter()
```

Comment:
I had to use progressutil.NewCopyProgressPrinter() this to avoid a panic with latest progress util.